### PR TITLE
Add text on APFS in macOS High Sierra

### LINKED ISSUE #5318: Default macOS filesystem changed from case-insensitive HFS+ to case-insensitive APFS
File: [docker-for-mac/osxfs.md](https://docs.docker.com/docker-for-mac/osxfs/), CC @gbarr01 

In the "Case sensitivity" section, it says the default macOS filesystem is `HFS+`. However, since macOS 10.12 High Sierra, the default changed to case-insensitive `APFS`.

Here is the official reference from Apple (go to "How does Apple File System handle filenames"):
https://developer.apple.com/library/content/documentation/FileManagement/Conceptual/APFS_Guide/FAQ/FAQ.html

The same issue occurs again in the section "Symlinks". In this case, the word `HFS+` could be deleted to solve the problem.

### PR DESCRIPTION
**Pull request by gbarr01**
Add information on the change to APFS in macOS High Sierra.

Fixes #5318

### CODE REVIEW

File: docker-for-mac/osxfs.md [hunk 1]
```diff
@@ -6,23 +6,29 @@ redirect_from:
...
-as HFS+ with case-sensitivity or reformat your OS root partition with HFS+ with
-case-sensitivity. We do not recommend reformatting your root partition as some
-Mac software dubiously relies on case-insensitivity to function.
+With Docker for Mac, file systems operate in containers in the same way as they
+operate in macOS. If a file system on macOS is case-insensitive, that behavior
+is shared by any bind mount from macOS into a container.
 
+On macOS Sierra and lower, the default file system is **HFS+**. On macOS High
+Sierra, the default file system is **APFS**. Both are case-insensitive by
+default but available in case-sensitive and  case-insensitive variants.
```
Comment (johndmulhausen) [inline docker-for-mac/osxfs.md]: whitespace

File: docker-for-mac/osxfs.md [hunk 2]
```diff
@@ -6,23 +6,29 @@ redirect_from:
...
+
+To get case-sensitive behavior from your bind mounts, you must either create and
+format a ramdisk or external volume as HFS+ or APFS with case-sensitivity, or
+else you must reformat your OS root partition with HFS+ or APFS with
```
Comment (johndmulhausen) [inline docker-for-mac/osxfs.md]: You can get away with removing the words "else you must"

File: docker-for-mac/osxfs.md [hunk 3]
```diff
@@ -6,23 +6,27 @@ redirect_from:
 title: File system sharing (osxfs)
 ---
 
-`osxfs` is a new shared file system solution, exclusive to Docker for Mac.
+`osxfs` is a new  shared file system solution, exclusive to Docker for Mac.
```
Comment (mistyhacks) [inline docker-for-mac/osxfs.md]: whitespace

File: docker-for-mac/osxfs.md [hunk 4]
```diff
@@ -6,23 +6,27 @@ redirect_from:
...
 
+On macOS Sierra and lower, the default file system is **HFS+**. On macOS High
+Sierra, the default file system is **APFS**. Both are case-insensitive by
+default but available in case-sensitive and case-insensitive variants.
+
+To get case-sensitive behavior, format the volume used in your bind mount as
+HFS+ or APFS with case-sensitivity. See the
+[APFS FAQ](https://developer.apple.com/library/content/documentation/FileManagement/Conceptual/APFS_Guide/FAQ/FAQ.html).
+
+We do not recommend reformatting your root partition as some Mac software relies
```
Comment (mistyhacks) [inline docker-for-mac/osxfs.md]: Maybe "Reformatting... is not recommended"?

Comment (gbarr01) [reply docker-for-mac/osxfs.md]: That was existing text but sure.

**Pull request merged**